docs/spec/relayer/Chain.tla

```diff
@@ -94,15 +119,62 @@ UpdateChainStore(chainID, datagrams) ==
     \* ICS 003: Channel updates
     LET channelUpdatedStore == ChannelUpdate(chainID, connectionUpdatedStore, datagrams) IN
 
+    \* ICS 004: Packet transmission
+    LET packetUpdatedStoreAndLogEntry == PacketUpdate(chainID, channelUpdatedStore, packetDatagrams, log) IN
+    LET packetUpdatedStore == packetUpdatedStoreAndLogEntry.chainStore IN
 
     \* update height
     LET updatedChainStore == 
-        IF /\ chainStore /= channelUpdatedStore
+        IF /\ chainStore /= packetUpdatedStore
            /\ chainStore.height + 1 \in Heights 
-        THEN [channelUpdatedStore EXCEPT !.height = chainStore.height + 1]
-        ELSE channelUpdatedStore
+        THEN [packetUpdatedStore EXCEPT !.height = chainStore.height + 1]
+        ELSE packetUpdatedStore
     IN
-    updatedChainStore
+
+    [chainStore |-> updatedChainStore, 
+     packetLog |-> packetUpdatedStoreAndLogEntry.packetLog]
+    
+(***************************************************************************
+ Packet acknowledgement operators
+ ***************************************************************************)    
+
+\* write acknowledgements to chain store
+WriteAcknowledgement(chain, packet) ==
+    \* if the acknowledgement for the packet has not been written
+    IF packet \notin chain.packetAcknowledgements
+    THEN \* write the acknowledgement to the chain store and remove 
+         \* the packet from the set of packets to acknowledge
+         LET packetAcknowledgement == 
+                AsPacketAcknowledgement(
+                    [channelID |-> packet.dstChannelID,
+                     sequence |-> packet.sequence,
+                     acknowledgement |-> TRUE]) IN
+         [chain EXCEPT !.packetAcknowledgements = 
+                            chain.packetAcknowledgements
+                            \union 
+                            {packetAcknowledgement},
+                       !.packetsToAcknowledge = 
+                            Tail(chain.packetsToAcknowledge)]                         
+    
+    \* remove the packet from the set of packets to acknowledge
+    ELSE [chain EXCEPT !.packetsToAcknowledge = 
+                            Tail(chain.packetsToAcknowledge)] 
+
+\* log acknowledgements to packet Log
+LogAcknowledgement(chain, log, packet) ==
```

Comment:
I am not sure if we need to model event log as it is not necessarily, i.e., current ICS18 is not using it. It might be useful as part of MBT, but in the initial version I would prefer having the minimal set of functionalities needed.